.github/workflows/build.yml

```diff
@@ -1,18 +1,51 @@
-name: Build Docker image
+name: Build and test Docker image
 
+# Run builds on main commits and when PRs are opened or updated
 on:
   push:
     branches:
-      - main
+    - main
+  pull_request:
```

Comment:
This is a slight change in behavior relative to the old configuration. We'll run the build when a PR is opened and when new commits are pushed to the branch after the PR is opened, but we won't run builds on a branch that doesn't have a corresponding PR.

I think this is desirable because we no longer run every build twice once you have opened a PR, and it should be easy enough to adapt to by just opening draft PRs aggressively.